// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTUnitDefinition";
option optimize_for = LITE_RUNTIME;

import public "BasketParams.proto";
import public "ContactEnums.proto";
import public "ContractEnums.proto";
import public "CurrencyParams.proto";
import public "EquityParams.proto";
import public "Nym.proto";
import public "Signature.proto";

message UnitDefinition {
    optional uint32 version = 1;
    optional string id = 2;
    optional string nymid = 3;
    optional Nym publicNym = 4;
    optional string shortName = 5;
    optional string terms = 6;
    optional string name = 7;
    optional string symbol = 8;
    optional UnitType type = 9;
    optional CurrencyParams currency = 10;
    optional EquityParams security = 11;
    optional BasketParams basket = 12;
    optional ContactItemType unitOfAccount = 13;
    optional Signature signature = 15;
}
